src/nvim/os/shell.c

```diff
@@ -37,6 +37,50 @@ typedef struct {
 #endif
 
 
+/// Process command string with 'shellxescape' (p_sxe) and 'shellxquote'
+/// (p_sxq)
+///
+/// @param cmd Command string
+/// @return NULL if `cmd` is NULL. Otherwise, a newly allocated command string.
+///         It must be freed with `xfree` when no longer needed.
+static char *shell_escape(const char *cmd)
```

Comment:
@brglng For macros FUNC_ATTR_FOO and REAL_FATTR_FOO check compiler documentation regarding `__attribute__(foo)`.